### PR TITLE
Fwinfo cleanup

### PR DESCRIPTION
**Pull request by oyvindronningstad**

### CODE REVIEW

**Comment by sigvartmh** [comment]
You have to rebase to get rid of the Kconfig issue it's fixed in master.

**Comment by oyvindronningstad** [comment]
> Changes looks good to me. Is there any reasoning for keeping the magic values so separated (value-range) ? I'm just curious mostly, I would mostly expect that related magic words would have a similar prefix/value-range but I do see that it could have some downsides.

Not quite sure what you are asking, but I split the magic number into fields to make it easier to give them different meaning. So fex. if you are changing to different HW, there is already a ready-made field you can change in Kconfig, you don't have to make a scheme for yourself. It's for convenience, and not meant to be parsed afterwards.

File: subsys/fw_info/fw_info.c [hunk 1]
```diff
@@ -154,8 +155,8 @@ SYS_INIT(check_ext_api_requests, POST_KERNEL, CONFIG_APPLICATION_INIT_PRIORITY);
 /** Value to write to the "valid" member of fw_info to invalidate the image. */
 #define INVALID_VAL 0xFFFF0000
 
-BUILD_ASSERT_MSG((INVALID_VAL & CONFIG_FW_INFO_VALID_VAL) \
-			!= CONFIG_FW_INFO_VALID_VAL, \
+BUILD_ASSERT_MSG((INVALID_VAL & CONFIG_FW_INFO_VALID_VAL)
+			!= CONFIG_FW_INFO_VALID_VAL,
```
Comment (nvlsianpu) [inline subsys/fw_info/fw_info.c]: `!=` should land in the first line

Comment (oyvindronningstad) [reply subsys/fw_info/fw_info.c]: Can I ask why? I find it more readable this way.

**Comment by oyvindronningstad** [comment]
Ready to merge.

**Pull request merged**